libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CoreBotAdapter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// CoreBotAdapter is a base BotAdapter that derives from CloudAdapter and implements OnTurnError handling. This class
+    /// also registers any middleware services that are contained in the middlewares collection.
+    /// </summary>
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes. This class is only instantiated via dependency injection so no direct reference is made.
+    internal class CoreBotAdapter : CloudAdapter
+#pragma warning restore CA1812 // Avoid uninstantiated internal classes
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoreBotAdapter"/> class.
+        /// </summary>
+        /// <param name="botFrameworkAuthentication">An instance of a BotFrameworkAuthentication.</param>
+        /// <param name="middlewares">The collection of middleware instances.</param>
+        /// <param name="logger">The logger instance.</param>
+        public CoreBotAdapter(
+            BotFrameworkAuthentication botFrameworkAuthentication,
+            IEnumerable<IMiddleware> middlewares,
+            ILogger<CoreBotAdapter> logger = null)
+            : base(botFrameworkAuthentication, logger)
+        {
+            // Pick up feature based middlewares such as telemetry or transcripts
+            foreach (IMiddleware middleware in middlewares)
+            {
+                Use(middleware);
+            }
+
+            OnTurnError = async (turnContext, exception) =>
```

Comment:
This would be another place where we want to let users customize what happens if there's an error in the bot. 
A developer may also want to be able to see the full exception while debugging a bot locally and we are hidding the stack trace here

Reply:
Instead of adding this class, could we add this as a default error handler to CloudAdapter? Then a user could still add a custom adapter if necessary and override OnTurnError. The method ServiceCollectionExtensions.RegisterCommonServices would query the services collection and only add CloudAdapter if a custom services is not found.

In StartUp:
services.AddBotRuntime(configuration);
services.UseBot<MyBot>();
services.UseBotHttpAdapter<customAdapter>(); <--optional